azkaban-exec-server/src/main/java/azkaban/dag/Status.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.dag;
+
+import com.google.common.collect.ImmutableSet;
+
+enum Status {
```

Comment:
should we have two separated status enum for dag and node each?

Reply:
Good question.

I thought about it.

The main factor that made me decide to use the same enum is to make testing easier. The same type can be put in the StatusChangeRecorder. 

That said, I can use string in the StatusChangeRecorder. Let's discuss more if it is worth refactoring.

Reply:
I will try it.